WooCommerce/Classes/ViewRelated/Dashboard/Coupons/MostActiveCouponsCard.swift

```diff
@@ -0,0 +1,254 @@
+                    .padding(.leading, Layout.padding)
+                    .padding(.vertical, Layout.hideIconVerticalPadding)
+            }
+            .disabled(viewModel.syncingData)
+        }
+    }
+
+    var couponsList: some View {
+        VStack(alignment: .leading, spacing: Layout.padding) {
+            // Header
+            HStack {
+                Text(Localization.coupons)
+                    .fontWeight(.bold)
+                    .subheadlineStyle()
+                Spacer()
+                Text(Localization.uses)
+                    .fontWeight(.bold)
+                    .subheadlineStyle()
+            }
+            .padding(.horizontal, Layout.padding)
+
+            // Rows
+            ForEach(viewModel.coupons) { item in
+                CouponDashboardRow(coupon: item, tapHandler: {
+                    viewModel.didSelectCoupon(item)
+                })
+            }
+            .redacted(reason: viewModel.syncingData ? [.placeholder] : [])
+            .shimmering(active: viewModel.syncingData)
+        }
+    }
+
+    var timeRangeBar: some View {
+        HStack {
+            AdaptiveStack(horizontalAlignment: .leading) {
+                Text(viewModel.timeRange.isCustomTimeRange ?
+                     Localization.custom : viewModel.timeRange.tabTitle)
+                .foregroundStyle(Color(.text))
+                .subheadlineStyle()
+
+                if viewModel.timeRange.isCustomTimeRange {
+                    Button {
+                        showingCustomRangePicker = true
+                    } label: {
+                        HStack {
+                            Text(viewModel.timeRangeText)
+                            Image(systemName: "pencil")
+                        }
+                        .foregroundStyle(Color.accentColor)
+                        .subheadlineStyle()
+                    }
+                } else {
+                    Text(viewModel.timeRangeText)
+                        .subheadlineStyle()
+                }
+            }
+            Spacer()
+            StatsTimeRangePicker(currentTimeRange: viewModel.timeRange) { newTimeRange in
+                if newTimeRange.isCustomTimeRange {
+                    showingCustomRangePicker = true
+                } else {
+                    viewModel.didSelectTimeRange(newTimeRange)
+                }
+            }
+            .disabled(viewModel.syncingData)
+        }
+    }
+
+    var viewAllCouponsButton: some View {
+        Button {
+            onViewAllCoupons(viewModel.siteID)
+        } label: {
+            HStack {
+                Text(Localization.viewAll)
+                Spacer()
+                Image(systemName: "chevron.forward")
+                    .foregroundStyle(Color(.tertiaryLabel))
+            }
+        }
+        .disabled(viewModel.syncingData)
+    }
+
+    func couponDetailView(coupon: Coupon) -> some View {
+        NavigationView {
```

Comment:
`NavigationView` is deprecated soon, please switch to `NavigationStack` instead, also to avoid issue when the view is displayed in a split view on large screens.

Reply:
I started using dashboard hosting view controller to present the coupon detail screen in 7ecea97489680952f5ea030c7f7ff100553e2c7c